### PR TITLE
ConnPool check  fd for bad conns

### PR DESCRIPTION
**Pull request by hidu**
it's work

Benchmark with Ping(No bad conns):

cpu: Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
Benchmark_Ping_before-8   	 4014712	       289.7 ns/op	     164 B/op	       4 allocs/op
Benchmark_Ping_after-8    	 4029973	       286.7 ns/op	     164 B/op	       4 allocs/op

Benchmark_Ping_before-8   	 4039963	       290.5 ns/op	     164 B/op	       4 allocs/op
Benchmark_Ping_after-8    	 4077626	       290.2 ns/op	     164 B/op	       4 allocs/op

Benchmark_Ping_before-8   	 4018032	       285.9 ns/op	     164 B/op	       4 allocs/op
Benchmark_Ping_after-8    	 4074387	       292.3 ns/op	     164 B/op	       4 allocs/op

### CODE REVIEW

**Comment by vmihailenco** [comment]
The build failure looks real. I guess it happens because `connCheck` detects bad conn and closes it...

**Comment by hidu** [comment]
got it,let me fix it

**Comment by monkey92t** [comment]
It seems to cause some test failures and we need to spend some time to resolve it.

**Comment by monkey92t** [comment]
I fixed the test error and it looks good now.

@vmihailenco what do you think?

**Review by monkey92t — APPROVED** [review]

File: pool_test.go [hunk 1]
```diff
@@ -87,8 +87,9 @@ var _ = Describe("pool", func() {
 		cn.SetNetConn(&badConn{})
 		client.Pool().Put(ctx, cn)
 
+		// connCheck will automatically remove damaged connections.
 		err = client.Ping(ctx).Err()
-		Expect(err).To(MatchError("bad connection"))
+		Expect(err).NotTo(HaveOccurred())
```
Comment (vmihailenco) [inline pool_test.go:92]: Nice!

**Comment by vmihailenco** [comment]
Looks good - thanks everyone :+1:

**Review by vmihailenco — APPROVED** [review]

**Comment by MrChenCode** [comment]
> it's work
> 
> Benchmark with Ping(No bad conns):
> 
> cpu: Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
> Benchmark_Ping_before-8 4014712 289.7 ns/op 164 B/op 4 allocs/op
> Benchmark_Ping_after-8 4029973 286.7 ns/op 164 B/op 4 allocs/op
> 
> Benchmark_Ping_before-8 4039963 290.5 ns/op 164 B/op 4 allocs/op
> Benchmark_Ping_after-8 4077626 290.2 ns/op 164 B/op 4 allocs/op
> 
> Benchmark_Ping_before-8 4018032 285.9 ns/op 164 B/op 4 allocs/op
> Benchmark_Ping_after-8 4074387 292.3 ns/op 164 B/op 4 allocs/op

My friend, can you give me your email address?

**Comment by hidu** [comment]
Thank you @monkey92t for completing my unfinished part！

**Pull request merged**